CMakeLists.txt

```diff
@@ -32,8 +32,20 @@ option(RE2C_BUILD_RE2RUST "Build re2rust executable (an alias for `re2c --lang r
 option(RE2C_BUILD_BENCHMARKS "Build benchmarks" OFF)
 option(RE2C_REGEN_BENCHMARKS "Regenerate C code for benchmarks" OFF)
 
+# check whether re2c is the root project
+if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
+    set(RE2C_IS_ROOT_PROJECT TRUE)
+else()
+    set(RE2C_IS_ROOT_PROJECT FALSE)
+endif()
+
+# test targets are enabled by default only if re2c is the root project
+option(RE2C_BUILD_TESTS "Build tests" "${RE2C_IS_ROOT_PROJECT}")
+
 # checks for programs
-find_package(Python3 3.7 REQUIRED COMPONENTS Interpreter)
+if(RE2C_REBUILD_DOCS OR RE2C_BUILD_TESTS)
```

Comment:
This should also include `RE2C_BUILD_BENCHMARKS` as they depend on  a few python scripts (to run the benchmarks).